deepmd/pd/utils/exclude_mask.py

```
@@ -0,0 +1,164 @@
+        """
+        nf, natom = atype.shape
+        return self.type_mask[atype].reshape([nf, natom]).to(atype.place)
+
+
+class PairExcludeMask(paddle.nn.Layer):
+    """Computes the type exclusion mask for atom pairs."""
+
+    def __init__(
+        self,
+        ntypes: int,
+        exclude_types: list[tuple[int, int]] = [],
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)